test/github-url.ts

```diff
@@ -15,6 +15,18 @@ test('branch', t => {
 	t.is(String(url), 'https://github.com/microsoft/TypeScript/tree/master');
 });
 
+test('branch with multiple slashes', t => {
+	const url = new GitHubURL('https://github.com/yakov116/TestR/tree/this/branch%2Fhas%2Fmany%2Fslashes');
+	t.is(url.user, 'yakov116');
+	t.is(url.repository, 'TestR');
+	t.is(url.route, 'tree');
+	t.is(url.branch, 'this/branch/has/many/slashes');
+	t.is(url.filePath, '');
```

Comment:
Let's also add a file path to ensure it's detected correctly. Is it possible?

Reply:
I think the link stops working this way as soon as the file path is added, but the code that **sets** `filePath: ''` is likely incorrect anyway.

Reply:
> I think the link stops working this way as soon as the file path is added

Yes, if you go to https://github.com/yakov116/TestR/tree/this/branch%2Fhas%2Fmany%2Fslashes/sub/README.md GitHub redirects to an URL with slashes